### PR TITLE
[Ember native classes] Transform code to use native classes

### PR DESCRIPTION
**Pull request by ssutar**
The code changes are after running codemods [ember-es6-class-codemod](https://github.com/scalvert/ember-es6-class-codemod)

This commit also adds following dependencies
- [Ember decorators](https://github.com/ember-decorators/ember-decorators)
- [Ember native class polyfills](https://github.com/pzuraq/ember-native-class-polyfill)

### CODE REVIEW

**Comment by rwjblue** [comment]
I’d generally rather see us update to ember-source@3.6.0 instead of using the native-class-polyfill. Can you try that?

**Comment by ssutar** [comment]
@rwjblue Updated with ember-source v3.6.0

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,11602 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (rwjblue) [inline yarn.lock]: Looks like this repo is using `package-lock.json` (not `yarn.lock`), mind deleting this and updating the package-lock.json?

**Comment by ssutar** [comment]
@rwjblue Using package-lock.json fails the build with following error:
```
6:01:19 PM: Requires Babel "^7.0.0-0", but was loaded with "6.26.3". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel.
```
However everything works fine with `yarn`

**Comment by ssutar** [comment]
Error summary from the log:

```
ERROR Summary:

  - broccoliBuilderErrorStack: [undefined]
  - codeFrame: [undefined]
  - errorMessage: Requires Babel "^7.0.0-0", but was loaded with "6.26.3". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel.
  - errorType: [undefined]
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
  - message: Requires Babel "^7.0.0-0", but was loaded with "6.26.3". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel.
  - name: Error
  - nodeAnnotation: [undefined]
  - nodeName: [undefined]
  - originalErrorMessage: [undefined]
  - stack: Error: Requires Babel "^7.0.0-0", but was loaded with "6.26.3". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel.
    at throwVersionError (/workspace/deprecation-app/node_modules/@babel/helper-plugin-utils/lib/index.js:65:11)
    at Object.assertVersion (/workspace/deprecation-app/node_modules/@babel/helper-plugin-utils/lib/index.js:13:11)
    at _default (/workspace/deprecation-app/node_modules/@babel/plugin-proposal-decorators/lib/index.js:35:7)
    at /workspace/deprecation-app/node_modules/@babel/helper-plugin-utils/lib/index.js:19:12
    at Function.memoisePluginContainer (/workspace/deprecation-app/node_modules/ember-auto-import/node_modules/babel-core/lib/transformation/file/options/option-manager.js:113:13)
    at Function.normalisePlugin (/workspace/deprecation-app/node_modules/ember-auto-import/node_modules/babel-core/lib/transformation/file/options/option-manager.js:146:32)
    at /workspace/deprecation-app/node_modules/ember-auto-import/node_modules/babel-core/lib/transformation/file/options/option-manager.js:184:30
    at Array.map (<anonymous>)
    at Function.normalisePlugins (/workspace/deprecation-app/node_modules/ember-auto-import/node_modules/babel-core/lib/transformation/file/options/option-manager.js:158:20)
    at OptionManager.mergeOptions (/workspace/deprecation-app/node_modules/ember-auto-import/node_modules/babel-core/lib/transformation/file/options/option-manager.js:234:36)
    at OptionManager.init (/workspace/deprecation-app/node_modules/ember-auto-import/node_modules/babel-core/lib/transformation/file/options/option-manager.js:368:12)
    at File.initOptions (/workspace/deprecation-app/node_modules/ember-auto-import/node_modules/babel-core/lib/transformation/file/index.js:212:65)
    at new File (/workspace/deprecation-app/node_modules/ember-auto-import/node_modules/babel-core/lib/transformation/file/index.js:135:24)
    at Analyzer.buildParserOptions (/workspace/deprecation-app/node_modules/ember-auto-import/js/analyzer.js:50:17)
    at new Analyzer (/workspace/deprecation-app/node_modules/ember-auto-import/js/analyzer.js:39:35)
    at AutoImport.analyze (/workspace/deprecation-app/node_modules/ember-auto-import/js/auto-import.js:41:24)
    at Object.toTree (/workspace/deprecation-app/node_modules/ember-auto-import/js/index.js:21:59)
    at /workspace/deprecation-app/node_modules/ember-cli-preprocess-registry/preprocessors.js:180:26
    at Array.forEach (<anonymous>)
    at processPlugins (/workspace/deprecation-app/node_modules/ember-cli-preprocess-registry/preprocessors.js:178:11)
    at Object.module.exports.preprocessJs (/workspace/deprecation-app/node_modules/ember-cli-preprocess-registry/preprocessors.js:171:10)
    at Class._getFastbootTree (/workspace/deprecation-app/node_modules/ember-cli-fastboot/index.js:183:32)
    at Class.treeForPublic (/workspace/deprecation-app/node_modules/ember-cli-fastboot/index.js:200:29)
    at Class._treeFor (/workspace/deprecation-app/node_modules/ember-cli/lib/models/addon.js:616:33)
    at Class.treeFor (/workspace/deprecation-app/node_modules/ember-cli/lib/models/addon.js:576:21)

=================================================================================

```

**Comment by ssutar** [comment]
Looks like the issue https://github.com/ef4/ember-auto-import/issues/119

**Pull request merged**